ansible/playbooks/adhoc/docker_storage_reinitialize/remediate_runc.yml

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/ansible-playbook
+---
+- name: "Playbook to un-schedule and drain drain compute nodes before remediating runc on them"
+  hosts: localhost
+  user: root
+
+  pre_tasks:
+  - fail:
+      msg: "This playbook requires {{item}} to be set."
+    when: item is not defined or item == ''
+    with_items:
+    - cli_master
+    - cli_node
+    - cli_ssh_name
+
+  tasks:
+    - name: "debug"
+      debug:
+        msg: "{{ item }}"
+      with_items:
+      -  "{{ cli_master }}"
+      -  "{{ cli_node }}"
+      -  "{{ cli_ssh_name }}"
+
+
+      # Test connectivity
+    - name: "Test ssh connection to the node from a bastion"
+      delegate_to: "root@{{ item }}"
+      shell: "uptime"
+      with_items:
+      -  "{{ cli_master }}"
+      -  "{{ cli_ssh_name }}"
+
+
+      # Begin master operations
+    - name: "Test oc get node from the primary master"
+      shell: "oc get node {{ cli_node }}"
+      delegate_to: "root@{{ cli_master }}"
+
+    - name: "Try to unschedule the node from the primary master"
+      shell: "oc adm manage-node {{ cli_node }} --schedulable=false"
+      delegate_to: "root@{{ cli_master }}"
+
+    - name: "Try to drain the node from the primary master"
+      shell: "oc adm drain {{ cli_node }} --force --ignore-daemonsets --delete-local-data"
+      delegate_to: "root@{{ cli_master }}"
+      ignore_errors: true
+
+
+    # Begin node operations
+    - name: "Disable docker before rebooting"
+      shell: "systemctl disable docker var-lib-docker.mount"
+      delegate_to: "root@{{ cli_ssh_name }}"
+
+    - name: "Reboot node to disable docker"
+      shell: "sleep 2 && shutdown -r now"
+      async: 5
+      poll: 0
+      delegate_to: "root@{{ cli_ssh_name }}"
+
+    - name: "Pause for 1 minute to allow the node to finish rebooting."
+      pause:
+        minutes: 1
+
+    - name: "Wait for the node to come back up"
+      shell: "uptime"
+      retry: 5
+      delay: 60
+      delegate_to: "root@{{ cli_ssh_name }}"
+      ignore_errors: true
+
+    - name: "Make sure docker is stopped and unmounted"
+      shell: "systemctl stop docker var-lib-docker.mount"
+      delegate_to: "root@{{ cli_ssh_name }}"
+
+
+    # Begin bastion operations
+    - name: "Start docker reinitialization playbook from a bastion"
+      shell: "ansible-playbook ops-docker-storage-reinitialize.yml -e cli_name={{ cli_ssh_name }} -e cli_confirm_run=yes"
+      delegate_to: localhost
+
+
+    # Continue node operations
+    - name: "Try to apply docker update first, then the rest of the pending updates, reenable docker"
+      shell: "{{ item }}"
+      with_items:
+      - "yum -y update --advisory=RHSA-2019:0304"
+      - "yum -y update --security"
```

Comment:
You do not want to run the whole list of unvetted security updates -- RHSA-2017:3188 will pull in unintended openshift packages on a handful of very old clusters.

Reply:
Good to know! I'm trying to get a solution in place for just the Starter / Pro / OSIO clusters at the moment, so it may be that I'll have to take a different route for the snowflakes.

Have you encountered any issues with this type of issue on the Starter / Pro / OSIO clusters as well?

Reply:
Use this, pulling the apply_advisory list from o-a-o-cve_update:
 yum update -y --advisory={{ apply_advisory | join(' --advisory=') }}"

Reply:
Closing this PR, it sounds like o-a-o is the right place to accommodate this suggestion.